frontend-react/src/contexts/SessionStorageTools.ts

```diff
@@ -2,47 +2,79 @@
  * right, but by no means is it a context. Feel free to move
  * it if you can think of a better spot. */
 
-import { AxiosRequestHeaders } from "axios";
-
-import { updateApiSessions } from "../network/Apis";
-
-const headersFromStoredSession = (): AxiosRequestHeaders => ({
-    Authorization: `Bearer ${getStoredOktaToken() || ""}`,
-    Organization: getStoredOrg() || "",
-});
+import {
+    MembershipSettings,
+    MembershipState,
+} from "../hooks/UseOktaMemberships";
 
 export enum GLOBAL_STORAGE_KEYS {
     GLOBAL_BASE = "global-context-",
     GLOBAL_ORG = "global-context-org",
     SENDER_NAME = "global-sender-name",
-    OKTA_ACCESS_TOKEN = "global-okta-token",
+    OKTA_ACCESS_TOKEN = "okta-token-storage", // set by okta
+    MEMBERSHIP_STATE = "global-membership-state",
+    ORGANIZATION_OVERRIDE = "global-organization-override",
 }
 
+const fetchJsonFromSession = (storageKey: string) => {
+    const storedString = sessionStorage.getItem(storageKey);
+    if (!storedString) {
+        return {};
+    }
+    try {
+        const sessionJson = JSON.parse(storedString);
+        return sessionJson;
+    } catch {
+        return {};
+    }
+};
+
+// temporary solution.
+// TODO: replace all occurrances of this with reads from SessionContext
 export function getStoredOktaToken(): string | undefined {
-    return (
-        sessionStorage.getItem(GLOBAL_STORAGE_KEYS.OKTA_ACCESS_TOKEN) ||
-        undefined
+    const tokenJsonString = sessionStorage.getItem(
```

Comment:
rather than manage storing the token ourselves, rely on okta's storage of the token, and just process it to our specifications as necessary